src/main/java/net/sacredlabyrinth/phaed/simpleclans/chat/handlers/SpyChatHandler.java

```diff
@@ -13,42 +14,37 @@
 import static net.sacredlabyrinth.phaed.simpleclans.chat.SCMessage.Source.DISCORD;
 import static net.sacredlabyrinth.phaed.simpleclans.chat.SCMessage.Source.SPIGOT;
 import static net.sacredlabyrinth.phaed.simpleclans.managers.SettingsManager.ConfigField;
-import static net.sacredlabyrinth.phaed.simpleclans.managers.SettingsManager.ConfigField.DISCORDCHAT_ENABLE;
-import static org.bukkit.Bukkit.getPluginManager;
 
 /**
- * Handles delivering messages from {@link Source#SPIGOT} and {@link Source#DISCORD} to internal spy chat.
+ * Handles delivering messages from {@link Source#SPIGOT} or {@link Source#DISCORD} to internal spy chat.
  */
-@SuppressWarnings("unused")
 public class SpyChatHandler implements ChatHandler {
 
     @Override
     public void sendMessage(SCMessage message) {
-        ConfigField configField = ConfigField.valueOf(String.format("%sCHAT_SPYFORMAT",
+        ConfigField formatField = ConfigField.valueOf(String.format("%sCHAT_SPYFORMAT",
                 message.getSource() == SPIGOT ? message.getChannel() : message.getSource()));
-        String format = settingsManager.getString(configField);
+        String format = settingsManager.getString(formatField);
         message.setContent(ChatUtils.stripColors(message.getContent()));
-
         String formattedMessage = chatManager.parseChatFormat(format, message);
 
-        List<ClanPlayer> onlinePlayers = getOnlineClanPlayers().stream()
-                .filter(player -> plugin.getPermissionsManager().has(player.toPlayer(), "simpleclans.admin.all-seeing-eye"))
-                .filter(player -> !player.isMuted())
-                .collect(Collectors.toList());
+        List<ClanPlayer> onlineSpies = getOnlineSpies();
 
-        onlinePlayers.removeAll(message.getReceivers());
-        onlinePlayers.forEach(receiver -> ChatBlock.sendMessage(receiver, formattedMessage));
+        // Don't send a duplicate message if a spy is inside the clan
+        onlineSpies.removeAll(message.getReceivers());
+        onlineSpies.forEach(receiver -> ChatBlock.sendMessage(receiver, formattedMessage));
     }
 
     @Override
     public boolean canHandle(SCMessage.Source source) {
-        return source == SPIGOT || (source == DISCORD && settingsManager.is(DISCORDCHAT_ENABLE) &&
-                getPluginManager().getPlugin("DiscordSRV") != null);
+        return source == SPIGOT || (source == DISCORD && chatManager.isDiscordHookEnabled());
     }
 
-    private List<ClanPlayer> getOnlineClanPlayers() {
-        return plugin.getClanManager().getAllClanPlayers().stream().
-                filter(cp -> cp.toPlayer() != null).
+    private List<ClanPlayer> getOnlineSpies() {
+        return Bukkit.getOnlinePlayers().stream().
```

Comment:
add a filter for Objects::requireNonNull, just in case
as Stream is suspected to be the cause 🤡